go.mod

```diff
@@ -7,57 +7,57 @@ replace sigs.k8s.io/cluster-api => sigs.k8s.io/cluster-api v1.5.3
 require (
 	github.com/alessio/shellescape v1.4.2
 	github.com/apparentlymart/go-cidr v1.1.0
-	github.com/aws/amazon-vpc-cni-k8s v1.15.4
-	github.com/aws/aws-lambda-go v1.41.0
-	github.com/aws/aws-sdk-go v1.44.332
+	github.com/aws/amazon-vpc-cni-k8s v1.16.0
+	github.com/aws/aws-lambda-go v1.44.0
+	github.com/aws/aws-sdk-go v1.48.10
 	github.com/awslabs/goformation/v4 v4.19.5
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/coreos/ignition v0.35.0
-	github.com/coreos/ignition/v2 v2.16.2
-	github.com/go-logr/logr v1.3.0
+	github.com/coreos/ignition/v2 v2.17.0
+	github.com/go-logr/logr v1.4.1
 	github.com/gofrs/flock v0.8.1
 	github.com/golang/mock v1.6.0
 	github.com/google/go-cmp v0.6.0
 	github.com/google/goexpect v0.0.0-20210430020637-ab937bf7fd6f
 	github.com/google/gofuzz v1.2.0
-	github.com/onsi/ginkgo/v2 v2.13.1
+	github.com/onsi/ginkgo/v2 v2.13.2
 	github.com/onsi/gomega v1.30.0
-	github.com/openshift-online/ocm-sdk-go v0.1.388
+	github.com/openshift-online/ocm-sdk-go v0.1.391
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.17.0
+	github.com/prometheus/client_golang v1.18.0
 	github.com/sergi/go-diff v1.3.1
 	github.com/spf13/cobra v1.8.0
 	github.com/spf13/pflag v1.0.6-0.20210604193023-d5e0c0615ace
-	golang.org/x/crypto v0.17.0
+	golang.org/x/crypto v0.18.0
 	golang.org/x/text v0.14.0
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.27.3
-	k8s.io/apiextensions-apiserver v0.27.3
-	k8s.io/apimachinery v0.27.3
-	k8s.io/cli-runtime v0.27.3
-	k8s.io/client-go v0.27.3
-	k8s.io/component-base v0.27.3
+	k8s.io/api v0.28.4
+	k8s.io/apiextensions-apiserver v0.28.3
+	k8s.io/apimachinery v0.28.4
+	k8s.io/cli-runtime v0.28.4
+	k8s.io/client-go v0.28.4
+	k8s.io/component-base v0.28.3
 	k8s.io/klog/v2 v2.100.1
-	k8s.io/utils v0.0.0-20230220204549-a5ecb0141aa5
-	sigs.k8s.io/aws-iam-authenticator v0.6.13
+	k8s.io/utils v0.0.0-20230505201702-9f6742963106
+	sigs.k8s.io/aws-iam-authenticator v0.6.17
 	sigs.k8s.io/cluster-api v1.5.3
 	sigs.k8s.io/cluster-api/test v1.5.3
-	sigs.k8s.io/controller-runtime v0.15.1
-	sigs.k8s.io/kustomize/api v0.13.4
+	sigs.k8s.io/controller-runtime v0.16.3
```

Comment:
Looks like this will require some signature updates. From CI:

```
./main.go:158:3: unknown field MetricsBindAddress in struct literal of type manager.Options
./main.go:163:3: unknown field SyncPeriod in struct literal of type manager.Options
./main.go:164:3: unknown field Namespace in struct literal of type manager.Options
./main.go:166:3: unknown field Port in struct literal of type manager.Options
./main.go:167:3: unknown field CertDir in struct literal of type manager.Options 
```

See the [release notes](https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.16.0)

Specifically https://github.com/kubernetes-sigs/controller-runtime/pull/2422 and https://github.com/kubernetes-sigs/controller-runtime/pull/2407